src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    const { extraData, keysToRemove } = action;
+
+    switch (action.type) {
```

Comment:
```suggestion
    const { 
      type, 
      extraData, 
      keysToRemove 
    } = action;

    switch (type) {
```
You can get type from destructuring too